// @uppy/angular/esm2022/lib/components/drag-drop/drag-drop.module.mjs
import { NgModule } from '@angular/core';
import { DragDropComponent } from './drag-drop.component';
import * as i0 from "@angular/core";
export const COMPONENTS = [DragDropComponent];
export class UppyAngularDragDropModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDragDropModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDragDropModule, declarations: [DragDropComponent], exports: [DragDropComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDragDropModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDragDropModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: COMPONENTS,
                    exports: COMPONENTS
                }]
        }] });
