// @uppy/angular/esm2022/lib/components/status-bar/status-bar.module.mjs
import { NgModule } from '@angular/core';
import { StatusBarComponent } from './status-bar.component';
import * as i0 from "@angular/core";
export const COMPONENTS = [StatusBarComponent];
export class UppyAngularStatusBarModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularStatusBarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularStatusBarModule, declarations: [StatusBarComponent], exports: [StatusBarComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularStatusBarModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularStatusBarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: COMPONENTS,
                    exports: COMPONENTS
                }]
        }] });
