// @uppy/angular/esm2022/lib/components/progress-bar/progress-bar.module.mjs
import { NgModule } from '@angular/core';
import { ProgressBarComponent } from './progress-bar.component';
import * as i0 from "@angular/core";
export const COMPONENTS = [ProgressBarComponent];
export class UppyAngularProgressBarModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularProgressBarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularProgressBarModule, declarations: [ProgressBarComponent], exports: [ProgressBarComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularProgressBarModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularProgressBarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: COMPONENTS,
                    exports: COMPONENTS
                }]
        }] });
