// @uppy/angular/esm2022/lib/components/dashboard-modal/dashboard-modal.module.mjs
import { NgModule } from '@angular/core';
import { DashboardModalComponent } from './dashboard-modal.component';
import * as i0 from "@angular/core";
export const COMPONENTS = [DashboardModalComponent];
export class UppyAngularDashboardModalModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDashboardModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDashboardModalModule, declarations: [DashboardModalComponent], exports: [DashboardModalComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDashboardModalModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDashboardModalModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: COMPONENTS,
                    exports: COMPONENTS
                }]
        }] });
