// @uppy/angular/esm2022/lib/components/dashboard-modal/dashboard-modal.component.mjs
import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
import Dashboard from '@uppy/dashboard';
import { Uppy } from '@uppy/core';
import { UppyAngularWrapper } from '../../utils/wrapper';
import * as i0 from "@angular/core";
export class DashboardModalComponent extends UppyAngularWrapper {
    constructor(el) {
        super();
        this.el = el;
        this.uppy = new Uppy;
        this.props = {};
        this.open = false;
    }
    ngOnInit() {
        this.onMount({
            id: 'angular:DashboardModal',
            inline: false,
            target: this.el.nativeElement
        }, Dashboard);
    }
    ngOnChanges(changes) {
        this.handleChanges(changes, Dashboard);
        // Handle dashboard-modal specific changes
        if (changes['open'] && this.open !== changes['open'].previousValue) {
            if (this.open && !changes['open'].previousValue) {
                this.plugin.openModal();
            }
            if (!this.open && changes['open'].previousValue) {
                this.plugin.closeModal();
            }
        }
    }
    ngOnDestroy() {
        this.uninstall();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: DashboardModalComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.1", type: DashboardModalComponent, selector: "uppy-dashboard-modal", inputs: { uppy: "uppy", props: "props", open: "open" }, usesInheritance: true, usesOnChanges: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: DashboardModalComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'uppy-dashboard-modal',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { uppy: [{
                type: Input
            }], props: [{
                type: Input
            }], open: [{
                type: Input
            }] } });
