// @uppy/angular/esm2022/lib/components/dashboard/dashboard.module.mjs
import { NgModule } from '@angular/core';
import { DashboardComponent } from './dashboard.component';
import * as i0 from "@angular/core";
export const COMPONENTS = [DashboardComponent];
export class UppyAngularDashboardModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDashboardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDashboardModule, declarations: [DashboardComponent], exports: [DashboardComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDashboardModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.1", ngImport: i0, type: UppyAngularDashboardModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: COMPONENTS,
                    exports: COMPONENTS
                }]
        }] });
